### PR TITLE
chore(gas_price_service): remove dependency on Config from fuel-core

### PR DESCRIPTION
**Pull request by rymnc**
> [!NOTE]
> This is PR 3/n in cleaning up the gas price service. 
> Please review https://github.com/FuelLabs/fuel-core/pull/2226  before getting to this one ;) 

## Description

Defines a custom configuration for the `GasPriceService` which is passed in on subservice initialization.

## Checklist
- [ ] Breaking changes are clearly marked as such in the PR description and changelog
- [ ] New behavior is reflected in tests
- [ ] [The specification](https://github.com/FuelLabs/fuel-specs/) matches the implemented behavior (link update PR if changes are needed)

### Before requesting review
- [ ] I have reviewed the code myself
- [ ] I have created follow-up issues caused by this PR and linked them here

### After merging, notify other teams

[Add or remove entries as needed]

- [ ] [Rust SDK](https://github.com/FuelLabs/fuels-rs/)
- [ ] [Sway compiler](https://github.com/FuelLabs/sway/)
- [ ] [Platform documentation](https://github.com/FuelLabs/devrel-requests/issues/new?assignees=&labels=new+request&projects=&template=NEW-REQUEST.yml&title=%5BRequest%5D%3A+) (for out-of-organization contributors, the person merging the PR will do this)
- [ ] Someone else?

### CODE REVIEW

File: crates/services/gas_price_service/src/ports.rs [hunk 1]
```diff
@@ -26,3 +26,26 @@ pub trait MetadataStorage: Send + Sync {
...
 }
+
+pub struct GasPriceServiceConfig {
+    pub min_gas_price: u64,
+    pub starting_gas_price: u64,
+    pub gas_price_change_percent: u64,
+    pub gas_price_threshold_percent: u64,
+}
+
+impl GasPriceServiceConfig {
```
Comment (acerone85) [inline crates/services/gas_price_service/src/ports.rs:37]: Is this going to be used in ports? 
If not, maybe it's best to move it either in a separate `config.rs` file or directly in `lib.rs`

Comment (acerone85) [reply crates/services/gas_price_service/src/ports.rs:37]: (comment refers both to the struct definition and implementation)

Comment (rymnc) [reply crates/services/gas_price_service/src/ports.rs:37]: well we define the implementation of the port (actually just the conversion from `Config` to `GasPriceServiceConfig`) in `crates/fuel-core/src/service/adapters/gas_price_adapters.rs` so to me it made sense to keep it all in one place :)

**Review by acerone85 — APPROVED** [review]

**Review by xgreenx — APPROVED** [review]

**Review by MitchTurner — APPROVED** [review]

**Comment by rymnc** [comment]
already merged into master

**Pull request closed without merging**